### PR TITLE
Instance tracking leftovers

### PR DESCRIPTION
**Pull request by mmedenjak**
- added jet and jet EE as products to be referenced from jet code
- add test for "server" mode

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/config/InstanceTrackingConfig.java [hunk 1]
```diff
@@ -256,7 +256,15 @@ public String getPropertyName() {
         /**
          * Hazelcast java enterprise client
          */
-        HAZELCAST_CLIENT_EE("Hazelcast Client Enterprise");
+        HAZELCAST_CLIENT_EE("Hazelcast Client Enterprise"),
+        /**
+         * Hazelcast jet open-source
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/config/InstanceTrackingConfig.java:261]: **J**et? :speak_no_evil:  :small_airplane:

**Review by blazember — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/config/InstanceTrackingConfig.java (hunk 1 above)
Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/config/InstanceTrackingConfig.java:261]: Oh no, what have I done? :)

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2087/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2087/Hazelcast-pr-builder-2087.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TransactionQueueTest.testIssue859And863_WhenQueuesAreOnSecondInstance:355->testIssue859And863:412->HazelcastTestSupport.assertTrueEventually:1350->HazelcastTestSupport.assertTrueEventually:1248 expected:<3000> but was:<2999>
[INFO] 
[ERROR] Tests run: 31217, Failures: 1, Errors: 0, Skipped: 976
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by mmedenjak** [comment]
Merging with 1 approval, change is simple and a test addition. Thank you for the review @blazember !

**Pull request merged**